openpgp/ecdh/x25519.go

```diff
@@ -11,13 +11,18 @@ import (
 	"io"
 	"math/big"
 
-	"golang.org/x/crypto/openpgp/aes/keywrap"
 	"golang.org/x/crypto/curve25519"
+	"golang.org/x/crypto/openpgp/aes/keywrap"
 	"golang.org/x/crypto/openpgp/internal/ecc"
 )
 
-func X25519GenerateParams(rand io.Reader) (priv [32]byte, x [32]byte, err error) {
-	var n, helper = new (big.Int), new (big.Int)
+// Generates a private-public key-pair.
+// 'priv' is a private key; a scalar belonging to the set
+// 2^{254} + 8 * [0, 2^{251}), in order to avoid the small subgroup of the
+// curve. 'pub' is simply 'priv' * G where G is the base point.
+// See https://cr.yp.to/ecdh.html and RFC7748, sec 5.
+func generateKeyPair(rand io.Reader) (priv [32]byte, pub [32]byte, err error) {
```

Comment:
Nit: this function is technically part of the public API, so we probably shouldn't remove it.

Other than that, LGTM :+1:

Reply:
As discussed, decided to unexport this function and provide a better name.